packages/walletconnect/src/utils.ts

```diff
@@ -66,3 +81,18 @@ export async function getBestUrl(urls: string[], timeout: number): Promise<strin
     })
   })
 }
+
+/**
+ * @param chains - An array of chain IDs
+ * @param defaultChainId - An ID to move to the front of the array (first element is always the default chain)
+ */
+export function orderToSetDefaultChain(chains: number[], defaultChainId: number) {
+  const ordered = [...chains]
+  const idx = ordered.indexOf(defaultChainId)
+  if (idx === -1) {
+    throw new Error(`Invalid chainId ${defaultChainId}. Make sure to include it in the "chains" array.`)
+  }
+  ordered.splice(idx, 1)
+  ordered.unshift(defaultChainId)
```

Comment:
I agree it's cleaner, but I don't like having to create a new array (so in this function, it's second time we do it). So, since it's a nit, I would stick to the way it's done unless you have strong opinion.